tests/testFlagHandler.py

```diff
@@ -0,0 +1,113 @@
+import lsst.meas.base
+import lsst.meas.base.tests
+import lsst.afw.table
+from lsst.meas.base.baseLib import MeasurementError
+from lsst.meas.base import FlagDefinition, FlagDefinitionVector, FlagHandler
+from lsst.meas.base.tests import (AlgorithmTestCase)
+
+class FlagHandlerTestCase(AlgorithmTestCase):
+    
+    def testFlagHandler(self):
+        control = lsst.meas.base.GaussianCentroidControl()
+        alg = lsst.meas.base.GaussianCentroidAlgorithm
+        schema = lsst.afw.table.SourceTable.makeMinimalSchema()
+        plugin = alg(control, 'test', schema)
+        cat = lsst.afw.table.SourceCatalog(schema)
+        schema = cat.getSchema()
+        subSchema = schema["test"]
+
+        # This is a FlagDefinition structure like a plugin might have
+        FAILURE = 0
+        FIRST = 1
+        SECOND = 2
+        flagDefs = [ FlagDefinition("General Failure", "general failure error"),
+            FlagDefinition("1st error", "this is the first failure type"),
+            FlagDefinition("2nd error", "this is the second failure type")
+        ]
+        fh = FlagHandler.addFields(schema, "test",
+            FlagDefinitionVector(flagDefs))
+
+        # Check to be sure that the FlagHandler was correctly initialized
+        for index, flagDef in enumerate(flagDefs):
+           assert(flagDef.name == flagDefs[index].name)
+           assert(flagDef.doc == flagDefs[index].doc)
+
+        catalog = lsst.afw.table.SourceCatalog(schema)
+
+        # Now check to be sure that all of the known failures set the bits correctly
+        record = catalog.addNew()
+        fh.handleFailure(record)
+        self.assertTrue(fh.getValue(record, 0))
+        self.assertFalse(fh.getValue(record, 1))
+        self.assertFalse(fh.getValue(record, 2))
+        record = catalog.addNew()
+
+        error = MeasurementError(fh.getDefinition(FAILURE).doc, FAILURE)
+        fh.handleFailure(record, error.cpp)
+        self.assertTrue(fh.getValue(record, 0))
+        self.assertFalse(fh.getValue(record, 1))
+        self.assertFalse(fh.getValue(record, 2))
+
+        record = catalog.addNew()
+        error = MeasurementError(fh.getDefinition(FIRST).doc, FIRST)
+        fh.handleFailure(record, error.cpp)
+        self.assertTrue(fh.getValue(record, 0))
+        self.assertTrue(fh.getValue(record, 1))
+        self.assertFalse(fh.getValue(record, 2))
+
+        record = catalog.addNew()
+        error = MeasurementError(fh.getDefinition(SECOND).doc, SECOND)
+        fh.handleFailure(record, error.cpp)
+        self.assertTrue(fh.getValue(record, 0))
+        self.assertFalse(fh.getValue(record, 1))
+        self.assertTrue(fh.getValue(record, 2))
+
+        record = catalog.addNew()
+        error = MeasurementError("Custom error message", FIRST)
+        fh.handleFailure(record, error.cpp)
+        self.assertTrue(fh.getValue(record, 0))
+        self.assertTrue(fh.getValue(record, 1))
+        self.assertFalse(fh.getValue(record, 2))
+
```

Comment:
It would actually be useful if you wrote a simple plugin that uses the flag handler. This could serve as an example for anyone going to use this in the future. Alternatively you could maybe just create a small example that uses the flag handler.